File: playground_series/s3e14/py/eda.py
import polars as pl
import seaborn as sns
import statistics

#apply default seaborn theme
sns.set_theme()

trn = pl.read_csv('./playground_series/s3e14/data/train.csv')
tst = pl.read_csv('./playground_series/s3e14/data/test.csv')

# Examine ------------------
trn.shape

trn.head()

#get column types
trn.dtypes
#ok so everything is numeric

# Target --------------
trn['yield'].mean()
trn['yield'].std()

sns.displot(trn, x='yield')

File: playground_series/s3e14/jl/eda.jl
using DataFrames
using CSV
using Statistics
using Plots

trn = CSV.read("./playground_series/s3e14/data/train.csv", DataFrame)
tst = CSV.read("./playground_series/s3e14/data/test.csv", DataFrame)

#Examine shape

size(trn)

# get element types
show(eltype.(eachcol(trn)))

#check missingness
show(mapcols(x -> sum(ismissing, x), trn), allcols = true)

# Explore Target --------------

mean(trn.:yield)

std(trn.:yield)

histogram(trn.:yield)
#slight left skew

# Explore Predictors -----------

#getting predictors
X = select(trn, Not(["id", "yield"]))

